c10/core/DispatchKey.h

```diff
@@ -193,6 +197,8 @@ enum class DispatchKey : uint16_t {
   // intended for out of tree use; tested by aten/src/ATen/test/rng_test.cpp
   CustomRNGKeyId,
 
+  // TODO: Make Mkldnn a functionality key, so we can give it Meta
```

Comment:
Oh welp, will mkldnn need their own meta kernel implementations?

I guess making mkldnn a functionality key is the way to go to save on dispatch key space. Feels a little bad having `MkldnnCUDA/XLA/etc` as "valid" keys that you can register things to though. I guess we can just register error kernels to them or something.

Reply:
yes they will, as it's a different layout with different rules